app/src/main/kotlin/at/bitfire/davdroid/sync/Syncer.kt

```diff
@@ -145,14 +142,14 @@ abstract class Syncer<StoreType: LocalDataStore<CollectionType>, CollectionType:
     internal fun updateCollections(
         provider: ContentProviderClient,
         localCollections: List<CollectionType>,
-        dbCollections: Map<HttpUrl, Collection>
+        dbCollections: Map<Long, Collection>
     ): List<CollectionType> {
         // create mutable copies of input
         val updatedLocalCollections = localCollections.toMutableList()
         val newDbCollections = dbCollections.toMutableMap()
 
         for (localCollection in localCollections) {
-            val dbCollection = dbCollections[localCollection.collectionUrl?.toHttpUrlOrNull()]
+            val dbCollection = dbCollections[localCollection.dbCollectionId]
```

Comment:
Just noticed that if the `localCollection.dbCollectionId` is null (corrupted local collection), this becomes

~~~Kotlin
val dbCollection = dbCollections[null]
~~~

although the map is a `Map<Long, Collection>` and not a `Map<Long?, Collection>`.

For some reason this seems to work, but I think it's to explicit enough to be clear. We should handle this case explicitly (and explicitly make `dbCollection == null` in this case).

Reply:
Right. It's valid and kind of shorthand for: 
`val dbCollection = dbCollections.getOrDefault(localCollection.dbCollectionId, null)`

See https://kotlinlang.org/docs/map-operations.html#retrieve-keys-and-values

But I agree, it might be too implicit (Android Studio is not suggesting it either).